pkg/chartutil/chartfile.go

```diff
@@ -33,7 +33,7 @@ func LoadChartfile(filename string) (*chart.Metadata, error) {
 		return nil, err
 	}
 	y := new(chart.Metadata)
-	err = yaml.Unmarshal(b, y)
+	err = yaml.UnmarshalStrict(b, y)
```

Comment:
Unfortunately, this will break non-lint usages of `LoadChartFile` which might have non-strict yaml (e.g. installing a chart). Helm maintains strong backwards compatibility guarantees in order to avoid breaking existing users.

(And to note `LoadChartfile` itself will follow API compatibility guarantees, and can't have its signature changed)

Reply:
Thanks for the update! I think the following preserves backward compatibility (of the other helm commands, as well as `helm lint` without the --strict option):
 - add a separate function `StrictLoadChartfile` to pkg/chartutil
 - use this in a WARNING-level linter rule in pkg/lint
 
 Would that be acceptable?

Reply:
Have updated this PR according to the above comment - this catches duplicate keys (and @Brightside56's error) at WARNING-level. Could I have another review @gjenkins8?